src/panic.md

```diff
@@ -16,14 +16,42 @@ There are also language features that provide a level of control over panic beha
 > [!NOTE]
 > The standard library provides the capability to explicitly panic via the [`panic!` macro][panic!].
 
+r[panic.runtime]
+## Panic runtimes
+
+The actual behavior and implementation of a panic is controlled by the _panic runtime_. The panic runtime is a handler linked into the output which provides the necessary implementation for panicking.
+
+> [!NOTE]
+> The Rust standard library provides two panic runtimes: `panic_unwind` (which unwinds the stack and is potentially recoverable) and `panic_abort` (which aborts the process and is non-recoverable). The default runtime depends on the target platform, but is generally `panic_unwind` on platforms with native support for C++ exceptions.
+
+> [!NOTE]
+> The panic runtime can be chosen in `rustc` with the [`-C panic`] CLI flag when building any crate type except an rlib.
+
+> [!NOTE]
+> When compiling code that is guaranteed to be linked to a non-recoverable panic runtime, the optimizer may assume that unwinding across Rust frames is impossible, which can result in both code-size and runtime speed improvements.
+
+See also the [`panic_handler` attribute](runtime.md#the-panic_handler-attribute) which can be used to change the behavior of panics.
+
+r[panic.strategy]
+## Panic strategy
+
+r[panic.strategy.intro]
+The _panic strategy_ defines the kind of panic runtime that a crate is built to support.
+
+> [!NOTE]
+> The panic strategy can be chosen in `rustc` with the [`-C panic`] CLI flag.
+
+r[panic.strategy.mixed]
+When linking with the `unwind` runtime, all crates must be built with the `unwind` strategy.
```

Comment:
This is redundant with `link.unwinding.prohibited`.

Reply:
> (To read that in a different way, I could say that these rules don't apply when using rustc since the section repeatedly talks about this being relevant to non-rustc linkers.)

Yeah and that is basically correct; if you use rustc it will check those rules so you don't have to worry about them.

But yeah I am open to rewording this, I can see how it can be confusing.